### PR TITLE
Fix SW net to ocean

### LINKED ISSUE #1768: SW net to ocean is too large at low sun angles
## Description

The SW net sent to the ocean in CMEPS uses a constant ocean albedo of 0.06 for the ocean fraction. This replicated the existing NEMS mediator functionality at the time we switched to CMEPS, but results in too much SW net to the ocean when the albedo > 0.06 due to a low sun angle.

This is an issue with both the DATM application as well as the fully coupled. It is not an issue for the HAFS model because HAFS is able to use SWnet from the ATM since there is no time-varying ice fraction.

## Additional context

When running fully coupled, the SW net to the ocean depends on both the time-varying ice-fraction, which determines the SW penetrating through the sea ice into the ocean, as well the SW net at the open ocean surface. Since the ocean runs at a slower coupling frequency than ATM-ICE, this is a mean value over the slow coupling frequency. That is

``SWnet to ocean = mean (swnet ocean * ofrac) + mean (ifrac * sw penetration)`` 

Currently, in UFS the swup over the ocean = (1-.06) * sw down. At low sun angles, when albedo>0.06, this over-estimates the open water SWnet.

Note that we cannot use just the SWnet provided by the ATM because we need to weight by the time-varying ice fraction. Otherwise, the value sent to the ocean would be

``SWnet to ocean = mean(swnet) * mean(ofrac) + mean(ifrac) * mean(sw penetration)``

To correctly account for both sun angle and ice-fraction, we need to calculate the ocean albedo at the end of each fast coupling step and then accumulate the net SW from both ATM and ICE. The ocean then receives the mean net SW to the ocean at its coupling frequency

The code to implement an ocean albedo phase is already present in the CMEPS code. It estimates ocean albedo as 

```
                ocnalb%anidr(n) = (.026_r8/(cosz**1.7_r8 + 0.065_r8)) +   &
                                  (.150_r8*(cosz         - 0.100_r8 ) *   &
                                  (cosz - 0.500_r8 ) * (cosz - 1.000_r8 )  )
                ocnalb%avsdr(n) = ocnalb%anidr(n)
                ocnalb%anidf(n) = albdif
                ocnalb%avsdf(n) = albdif

```

with albdif= 0.06 and albdir = 0.07. 

This appears to be the same formula used in ccpp/physics/physics/GFS_surface_generic_post.F90, which I think is where the albedo is being calculated for the global case.

```
              ocalnirbm_cpl = max(albdf, 0.026_kind_phys/(xcosz_loc**1.7_kind_phys+0.065_kind_phys)     &
       &                       + 0.15_kind_phys * (xcosz_loc-0.1_kind_phys) * (xcosz_loc-0.5_kind_phys) &
       &                       * (xcosz_loc-one))
```

with albdf = 0.06 (It isn't clear why the bm albedo is using the diffuse value here as a max, and not the direct value as a max).

The use of the max function is not present in the current CMEPS albedo code, but can easily be set as a configurable option so that the code still works for the CESM case. 

The ocean albedo calculation occurs with in a new ocnalb run phase which occurs after the ATM and ICE both run and before the fields are accumulated for the ocean. The SWnet for the open ocean is then 

SW net to ocean = Sw from atm * (1-ifrac) * (1-albedo) + ifrac * SW penetration

This should result in a lower SW net to ocean in the mean. 

I will post figures for a coupled and DATM case in the next few days.

### PR DESCRIPTION
**Pull request by DeniseWorthen**
## Description

Implements the ocean albedo calculation in CMEPS to determine SWnet sent to ocean. 

A configuration variable (``ocean_albedo_limit``) is added to the mediator attributes and set = 0.06 in order to match the behaviour of FV3 (see ccpp/physics/physics/GFS_surface_generic_post.F90). 

Two additional configuration variables (``flux_albav = true``, ``albdir = 0.06``) can be used to mimic the current treatement of SW net to the ocean even with the use of the mediator ocean albedo calculation.

### Input data additions/changes
- [x] No changes are expected to input data.
- [ ] There will be new input data. 
- [ ] Input data will be updated. 

### Anticipated changes to regression tests:
- [ ] No changes are expected to any regression test. 
- [x] Changes are expected to the following tests: 

[RegressionTests_hera.log](https://github.com/ufs-community/ufs-weather-model/files/12363032/RegressionTests_hera.log)

All global DATM and Coupled RTs will need new baselines. HAFS baselines do not change since the HAFS application uses SWnet from the ATM. The ``atmwav_control_noaero_p8_intel`` reproduces all files except for the WAV restart file. 

<details><summary>Failed tests</summary>

```sh
139:Test 001 cpld_control_p8_mixedmode_intel FAIL
210:Test 002 cpld_control_gfsv17_intel FAIL
282:Test 003 cpld_control_p8_intel FAIL
354:Test 005 cpld_control_qr_p8_intel FAIL
414:Test 007 cpld_2threads_p8_intel FAIL
474:Test 008 cpld_decomp_p8_intel FAIL
534:Test 009 cpld_mpi_p8_intel FAIL
606:Test 010 cpld_control_ciceC_p8_intel FAIL
666:Test 011 cpld_control_c192_p8_intel FAIL
721:Test 013 cpld_bmark_p8_intel FAIL
792:Test 015 cpld_control_noaero_p8_intel FAIL
861:Test 016 cpld_control_nowave_noaero_p8_intel FAIL
921:Test 017 cpld_debug_p8_intel FAIL
980:Test 018 cpld_debug_noaero_p8_intel FAIL
1049:Test 019 cpld_control_noaero_p8_agrid_intel FAIL
1106:Test 020 cpld_control_c48_intel FAIL
1178:Test 021 cpld_control_p8_faster_intel FAIL
1249:Test 022 cpld_control_pdlib_p8_intel FAIL
1308:Test 025 cpld_debug_pdlib_p8_intel FAIL
4901:Test 158 datm_cdeps_control_cfsr_intel FAIL
4914:Test 160 datm_cdeps_control_gefs_intel FAIL
4927:Test 161 datm_cdeps_iau_gefs_intel FAIL
4940:Test 162 datm_cdeps_stochy_gefs_intel FAIL
4953:Test 163 datm_cdeps_ciceC_cfsr_intel FAIL
4966:Test 164 datm_cdeps_bulk_cfsr_intel FAIL
4979:Test 165 datm_cdeps_bulk_gefs_intel FAIL
4995:Test 166 datm_cdeps_mx025_cfsr_intel FAIL
5011:Test 167 datm_cdeps_mx025_gefs_intel FAIL
5022:Test 168 datm_cdeps_multiple_files_cfsr_intel FAIL
5035:Test 169 datm_cdeps_3072x1536_cfsr_intel FAIL
5048:Test 170 datm_cdeps_gfs_intel FAIL
5061:Test 171 datm_cdeps_debug_cfsr_intel FAIL
5074:Test 172 datm_cdeps_control_cfsr_faster_intel FAIL
5248:Test 176 atmwav_control_noaero_p8_intel FAIL
7330:Test 244 cpld_control_p8_gnu FAIL
7399:Test 245 cpld_control_nowave_noaero_p8_gnu FAIL
7459:Test 246 cpld_debug_p8_gnu FAIL
7530:Test 247 cpld_control_pdlib_p8_gnu FAIL
7589:Test 248 cpld_debug_pdlib_p8_gnu FAIL
7602:Test 249 datm_cdeps_control_cfsr_gnu FAIL
```
</details>

## Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [x] CMEPS
- [ ] CMakeModules
- [ ] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [ ] none

## Commit Queue Checklist:

- [x] Link PR's from all sub-components involved
- [x] Confirm reviews completed in sub-component PR's
- [x] Add all appropriate labels to this PR.
- [x] Run full RT suite on either Hera/Cheyenne with both Intel/GNU compilers
- [x] Add list of any failed regression tests to "Anticipated changes to regression tests" section.

## Linked PR's and Issues:

- depends on #1752
- depends on #1776 
- fixes https://github.com/ufs-community/ufs-weather-model/issues/1768
- depends on https://github.com/NOAA-EMC/CMEPS/pull/92
- related to https://github.com/ufs-community/ufs-weather-model/issues/1767
## Testing Day Checklist:

- [ ] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.

### Testing Log (for CM's):
- RDHPCS
  - Intel
    - [x] Hera
    - [x] Orion
    - [x] Jet
    - [x] Gaea
    - [x] Cheyenne
  - GNU
    - [x] Hera
    - [x] Cheyenne
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- CI
  - [x] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
This is still WIP. I accidentally marked it as ready. It is not ready.

File: tests/parm/nems.configure.cpld.IN [hunk 1]
```diff
@@ -104,6 +104,7 @@ runSeq::
      MED med_phases_post_ice
      WAV -> MED :remapMethod=redist
      MED med_phases_post_wav
+     MED med_phases_ocnalb_run
```
Comment (junwang-noaa) [inline tests/parm/nems.configure.cpld.IN:107]: Are the changes backward compatible? Can you create an global-workflow issue to let people know the nems.configure changes for the coupled model?

Comment (DeniseWorthen) [reply tests/parm/nems.configure.cpld.IN:107]: Yes, I've created issue https://github.com/NOAA-EMC/global-workflow/issues/1811

**Review by junwang-noaa — DISMISSED** [review]

**Comment by FernandoAndrade-NOAA** [comment]
Jenkins ci logs are attached, apologies for the delay Jenkins had to be rerun on this PR due an issue with staging the test data, should be resolved now.
[ufs-weather-model » ort-docker-pipeline » PR-1773#1 Console [Jenkins].txt](https://github.com/ufs-community/ufs-weather-model/files/12434908/ufs-weather-model.ort-docker-pipeline.PR-1773.1.Console.Jenkins.txt)

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by zach1221 — APPROVED** [review]

**Review by jiandewang — CHANGES_REQUESTED** [review]
approve based on previous 8 mini testing results and today's RT test

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**